ruby/extractor/src/main.rs

```diff
@@ -39,6 +43,25 @@ fn num_codeql_threads() -> usize {
     }
 }
 
+lazy_static! {
+    static ref CP_NUMBER: regex::Regex = regex::Regex::new("cp([0-9]+)").unwrap();
+}
+
+fn encoding_from_name(encoding_name: &str) -> Option<&(dyn encoding::Encoding + Send + Sync)> {
+    match encoding::label::encoding_from_whatwg_label(&encoding_name) {
+        Some(e) => return Some(e),
+        None => {
+            if let Some(cap) = CP_NUMBER.captures(&encoding_name) {
+                return encoding::label::encoding_from_windows_code_page(
+                    str::parse(cap.get(1).unwrap().as_str()).unwrap(),
+                );
+            } else {
+                return None;
+            }
+        }
+    }
```

Comment:
We can make a few simplifications here to make it more Rust-y.

```suggestion
    match encoding::label::encoding_from_whatwg_label(encoding_name) {
        s @ Some(_) => s,
        None => CP_NUMBER.captures(encoding_name).and_then(|cap| {
            encoding::label::encoding_from_windows_code_page(
                str::parse(cap.get(1).unwrap().as_str()).unwrap(),
            )
        }),
    }
```

1.  No need to `return x;` when you can just write `x`.
2. `s @ Some(_) => s` lets you avoid destructuring and recreating the `Some(...)` value (not a big problem, I admit).
3. What you wrote in the `None` case can be written more succinctly with `Option::and_then`.